lib/omniauth/strategies/clever.rb

```diff
@@ -45,17 +45,73 @@ def callback_phase
       uid{ raw_info['data']['id'] }
 
       info do
-        { :user_type => raw_info['type'] }.merge! raw_info['data']
+        { :user_type => user_info['type'] }.merge! raw_info['data']
       end
 
       extra do
         {
           'raw_info' => raw_info
         }
       end
+      #  {
+      #       "type": "teacher",
+      #       "data": {
+      #         "id": "5798dc36fd5e25eb150010b5",
+      #         "district": "573f84ce403fe40100000fb7",
+      #         "type": "teacher"
+      #       },
+      #       "links": [
+      #         {
+      #           "rel": "self",
+      #           "uri": "/me"
+      #         },
+      #         {
+      #           "rel": "canonical",
+      #           "uri": "/v2.0/teachers/5798dc36fd5e25eb150010b5"
+      #         },
+      #         {
+      #           "rel": "district",
+      #           "uri": "/v2.0/districts/573f84ce403fe40100000fb7"
+      #         }
+      #       ]
+      #     }
+      def user_info
+        @user_info ||= access_token.get('/v2.0/me').parsed
+      end
+
+      #Should return "/v2.0/teachers/5798dc36fd5e25eb150010b5" from example above
+      def extract_canonical_uri
+         user_info['links'].find { |link| link['rel'] == 'canonical' }['uri']
+      end
 
+      # {
+      #   "data": {
+      #     "created": "2021-08-11T04:51:02.722Z",
+      #     "district": "573f84ce403fe40100000fb7",
+      #     "email": "carrie.davros@jeffco.k12.co.us",
```

Comment:
is this fake data? It just reads disconcertingly like a legit set of PII user info

Reply:
Yea it absolutely was. I had put it there as a reminder of the object and then never went back to edit. Thank you my bad. Updated. @sdawson